server/src/main/java/com/cloud/network/guru/ControlNetworkGuru.java

```diff
@@ -166,18 +167,25 @@ public boolean release(NicProfile nic, VirtualMachineProfile vm, String reservat
         assert nic.getTrafficType() == TrafficType.Control;
         HypervisorType hType = vm.getHypervisorType();
         if ( ( (hType == HypervisorType.VMware) || (hType == HypervisorType.Hyperv) )&& isRouterVm(vm)) {
+            // for now place this in the vmware specific part, but it miught be more generic and move up two or three lines
+            if (!VirtualNetworkApplianceManager.RemoveNicsOnStop.valueIn(vm.getVirtualMachine().getDataCenterId())) {
+                if (s_logger.isDebugEnabled()) {
+                    s_logger.debug(String.format("not releasing %s\n\t from %s\n\t with reservationId %s.", nic, vm, reservationId));
```

Comment:
why \n\t is used here (and other lines)?

Reply:
debugging aid, easier to comare objects in the logs this way. Can remove when finished.

Reply:
```suggestion
                    s_logger.debug(String.format("not releasing %s from %s with reservationId %s.", nic, vm, reservationId));
```